exporter/signalfxexporter/translation/translator.go

```diff
@@ -87,19 +188,66 @@ func createDimensionsMap(rules []Rule) map[string]string {
 	return nil
 }
 
-func (mp *MetricTranslator) TranslateDataPoints(sfxDataPoints []*sfxpb.DataPoint) {
+func (mp *MetricTranslator) TranslateDataPoints(logger *zap.Logger, sfxDataPoints []*sfxpb.DataPoint) []*sfxpb.DataPoint {
+	processedDataPoints := sfxDataPoints
+
 	for _, tr := range mp.rules {
```

Comment:
It would be useful to do some benchmarking to understand how much this impacts the performance. If it is significant we could keep one consolidated map to check dp.Metric against so that we don't have to do multiple checks per data point. We could eliminate some of the checks by doing that.
Fine for now, let's just keep an eye on the performance.

Reply:
We can try the consolidated map, but I'm not sure if it'll allow us to make "copy metric" -> "some modifications on the copy" to introduce the additional "compatibility" metrics